cli/args/mod.rs

```diff
@@ -122,6 +122,14 @@ impl RootConfig {
     )
   }
 
+  /// Removes the import map from the configuration.
+  pub fn remove_import_map(&mut self) {
```

Comment:
Oh? This seems a bit strange to me, we are we modifying configuration from the process?

Reply:
I need to ensure that the import map is not used for anything. An easy way to do that is to modify this configuration to not specify it and then use this mutated configuration that doesn't specify an import map to build everything.

Actually, maybe `RootOptions` might have been a better name than `RootConfig`?

Reply:
I'll change this name to `clear_import_map`

Reply:
> I need to ensure that the import map is not used for anything. An easy way to do that is to modify this configuration to not specify it and then use this mutated configuration that doesn't specify an import map to build everything.

Ok 👍 

> Actually, maybe `RootOptions` might have been a better name than `RootConfig`?

Yes, I think that might be a better name. Maybe `CliOptions` would be better?

Reply:
> Yes, I think that might be a better name. Maybe CliOptions would be better?

Ok, I like `CliOptions`. I can do another PR after this one that renames it.

Reply:
By the way, this lazily constructs the results by hiding the flags and config file inside the struct, but perhaps it would be more straightforward if this resolved and inline everything necessary when it's constructed. The one issue is that sometimes people might want to know if something came from a flag or from the config file.

Reply:
> By the way, this lazily constructs the results by hiding the flags and config file inside the struct, but perhaps it would be more straightforward if this resolved and inlined everything necessary when it's constructed. The one issue is that sometimes people might want to know if something came from a flag or from the config file.

I think it should evaluate eagerly. Can you suggest one such situation?

Reply:
I'll see when I go to work on it. I don't think it's necessary, but this was a decent stepping stone.

Reply:
So, one nice thing about doing it lazily is that we can surface errors only when necessary and can ignore errors in certain scenarios. For example, only displaying task errors when running a deno task or when using watch we could display an error about a certain part not working, but still continue execution.

Reply:
It already works lazily. Changing it to not be lazy could open a can of worms of more errors happening.

Reply:
@bartlomieju I added an "overrides" instead of clearing out the underlying structure. Let me know if this looks better than before.